src/Uno.UI/UI/Xaml/Media/VisualTreeHelper.cs

```diff
@@ -444,11 +444,176 @@ internal static IReadOnlyList<_View> ClearChildren(UIElement view)
 			return default;
 		}
 
+#if __SKIA__ // optimized implementation for skia that uses visuals directly
+		/// <param name="position">The absolute position relative to the window origin</param>
+		private static (UIElement? element, Branch? stale) SearchDownForTopMostElementAt(
```

Comment:
Please do not clone it ... if I'm right the difference is only on how we resolve the `clippingBounds` and the `renderingBounds`, those could be extracted in a platform specific method. Creating a complete separate implementation will make it more complex to maintain and diverge.

Reply:
It was too messy when everything was in one big function, but oddly enough, it looks quite pleasant after merging them again.